scripts/compiling.py

```diff
@@ -0,0 +1,322 @@
+			msg+=formatProperty(prop)
+
+		# generate each .msg file and add it to the list
+		if topic.fileName is None:
+			fileName = topic.name.title()+".msg"
+			msgFiles.append(topic.name.title())
+		else:
+			fileName = topic.fileName+".msg"
+			msgFiles.append(topic.fileName)
+
+		text_file = open(fileName, "w")
+		text_file.write(msg)
+		text_file.close()
+
+def formatProperty(prop):
+	if prop.fileName is None:
+		type = dataTypes[prop.type]
+	else:
+		type = msgPkgName+"/"+prop.fileName
+	return type+("[{}] ".format(prop.length if prop.length is not None else "") if prop.array else " ")+prop.name+((" # "+prop.desc) if prop.desc is not None else "")+"\n\n"
+
+def main(module):
+
+	messages = False
+	services = False
+	dependency = False
+
+	# default platform
+	plat = "ros2"
+
+	# map the datatypes
+	global dataTypes
+	dataTypes = {}
+	cwd = os.getcwd()
+	dataTree = ET.parse(cwd+"/models/dataMapping.xml")
+	dataRoot = dataTree.getroot()
+
+	# check for the platform
+	if any(platform.attrib.get("name") == plat for platform in dataRoot):
+		for platform in dataRoot.iter("platform"):
+			if platform.attrib.get("name") == plat:
+				for type in platform.iter("type"):
+					dataTypes[type.attrib.get("name")] = type.attrib.get("value")
+	else:
+		print "Chosen platform doesn't exist"
+		sys.exit(1)
+
+	# parse the templates necessary for the package
+	os.chdir("templates")
+	with open('package.txt', 'r') as myfile:
+		pkg=myfile.read()
+
+	# insert the package's name and description in package.xml's content
+	global msgPkgName
+	msgPkgName = "hrim_"+module.type+"_"+module.name+"_msgs"
+	msgPkg = pkg.replace("%PKGNAME%", msgPkgName)
+	msgPkg = msgPkg.replace("%PKGDESC%", module.desc)
+	msgPkg = msgPkg.replace("%PKGBUILD%", "")
+	msgPkg = msgPkg.replace("%PKGEXEC%", "")
+
+	# insert the package's name and description in package.xml's content
+	srvPkg = pkg.replace("%PKGNAME%", "hrim_"+module.type+"_"+module.name+"_srvs")
+	srvPkg = srvPkg.replace("%PKGDESC%", module.desc)
+
+	with open('cmake.txt', 'r') as myfile:
+		makeFile=myfile.read()
+
+	# insert the package's name and description in CMakeLists.txt's content
+	msgMakeFile = makeFile.replace("%PKGNAME%", msgPkgName)
+	msgMakeFile = msgMakeFile.replace("%PKGFIND%", "")
+	msgMakeFile = msgMakeFile.replace("%PKGDEP%", "")
+
+	# insert the package's name and description in CMakeLists.txt's content
+	srvMakeFile = makeFile.replace("%PKGNAME%", "hrim_"+module.type+"_"+module.name+"_srvs")
+
+	os.chdir(cwd)
+
+	# if the container directory for the generated module doesn't exist, create it
+	if not os.path.exists(cwd+"/generated"):
+			os.mkdir("generated")
+	os.chdir("generated")
+	cwd = os.getcwd()
+
+	# if the module directory for the generated files doesn't exist, create it
+	if not os.path.exists(cwd+"/"+module.name):
+		os.mkdir(module.name)
+
+	os.chdir(module.name)
+	cwd = os.getcwd()
+
+	# list of files for CMakeLists.txt
+	global msgFiles
```

Comment:
?

Reply:
Not longer a global, a list of the current package's generated files to add into CMakeLists.txt.

A very rudimentary separation was made for https://github.com/erlerobot/HRIM/commit/3b1a2495924686ce848278f129195221fd2a3c3e into msgFiles (for the generic ones) and ownFiles (for the package's specific ones).